pkg/controller/launch/pool_manager.go

```diff
@@ -281,25 +283,54 @@ func (cpm ConcretePoolManager) unNeeded(nodes []openstack_kluster.Node) int {
 	return unneeded
 }
 
-func (cpm *ConcretePoolManager) healthyAndSchedulable() (healthy int, schedulable int) {
+func (cpm *ConcretePoolManager) healthyAndSchedulable(nodeIDs []string) (int, int, []string) {
 	nodeLister, err := cpm.nodeObservatory.GetListerForKluster(cpm.Kluster)
 	if err != nil {
-		return
+		return 0, 0, nil
 	}
+
 	nodes, err := nodeLister.List(labels.Everything())
 	if err != nil {
-		return
+		return 0, 0, nil
 	}
+
+	healthy := 0
+	schedulable := 0
+
 	for _, node := range nodes {
 		//Does the node belong to this pool?
 		if util.IsKubernikusNode(node.Name, cpm.Kluster.Spec.Name, cpm.Pool.Name) {
-			if !node.Spec.Unschedulable {
+			if node.Spec.Unschedulable {
+				if nodeIDs != nil && len(nodeIDs) > 0 {
+					id := strings.Replace(node.Spec.ProviderID, "openstack:///", "", 1)
+					pos := findString(nodeIDs, id)
```

Comment:
I found this [IndexFunc()](https://pkg.go.dev/golang.org/x/exp/slices#IndexFunc) which should do the job?

Reply:
Hmm, you need to pass a function for comparison, then it would be the same.

Reply:
Index would be the right one, no: https://cs.opensource.google/go/x/exp/+/c95f2b4c:slices/slices.go;l=106